### PR TITLE
BugFix 1048014: Ensured build belongs to linked BD

### PR DESCRIPTION
**Pull request by satinder-kharea**
BugFix 1048014: Added the checks to ensure that the build artifact belongs to linked build definition only.

### CODE REVIEW

File: src/Agent.Worker/Release/Artifacts/BuildArtifact.cs [hunk 1]
```diff
@@ -123,6 +125,8 @@ public IArtifactDetails GetArtifactDetails(IExecutionContext context, AgentArtif
                     AccessToken = accessToken,
                     Project = projectId.ToString(),
                     TfsUrl = new Uri(tfsUrl),
+                    DefinitionName = artifactDetails["DefinitionName"],
```
Comment (GitHubSriramB) [inline src/Agent.Worker/Release/Artifacts/BuildArtifact.cs]: check for existence of key before usage? if agent is new & service is old, may be the key wouldn't exist. in which case it is fine to not check for this existence.

File: src/Agent.Worker/Release/Artifacts/BuildArtifact.cs [hunk 2]
```diff
@@ -123,6 +125,8 @@ public IArtifactDetails GetArtifactDetails(IExecutionContext context, AgentArtif
...
+                    DefintionId = Convert.ToInt32(artifactDetails["DefinitionId"], CultureInfo.InvariantCulture)
```
Comment (GitHubSriramB) [inline src/Agent.Worker/Release/Artifacts/BuildArtifact.cs]: same as above

File: src/Misc/layoutbin/en-US/strings.json [hunk 3]
```diff
@@ -562,5 +562,6 @@
   "WindowsLogonAccountNameDescription": "User account to use for the service",
   "WindowsLogonPasswordDescription": "Password for the account {0}",
   "WorkFolderDescription": "work folder",
-  "Y": "Y"
+  "Y": "Y",
+  "ArtifactNotFromLinkedDefintion": "The build {0} does not belong to the linked build definition {1}"
```
Comment (GitHubSriramB) [inline src/Misc/layoutbin/en-US/strings.json]: you need to sort the string ids in alphabetical order. one of the l0 tests will catch this. pl. make sure you run l0 tests and they pass with this fix.

Comment (GitHubSriramB) [inline src/Misc/layoutbin/en-US/strings.json]: I think we use RM* prefix for our resources, pl. check.

File: src/Agent.Worker/Release/Artifacts/BuildArtifact.cs (hunk 1 above)
Comment (satinder-kharea) [inline src/Agent.Worker/Release/Artifacts/BuildArtifact.cs]: Done.

File: src/Agent.Worker/Release/Artifacts/BuildArtifact.cs (hunk 2 above)
Comment (satinder-kharea) [inline src/Agent.Worker/Release/Artifacts/BuildArtifact.cs]: Done.

File: src/Misc/layoutbin/en-US/strings.json (hunk 3 above)
Comment (satinder-kharea) [inline src/Misc/layoutbin/en-US/strings.json]: Done. Ensured that tests are passing.

Comment (satinder-kharea) [inline src/Misc/layoutbin/en-US/strings.json]: Not sure about this one, this artifact is not RM specific

File: src/Misc/layoutbin/en-US/strings.json [hunk 4]
```diff
@@ -20,6 +20,7 @@
   "ArtifactDownloadFailed": "Failed to download the artifact from {0}.",
   "ArtifactLocationRequired": "Artifact location is required.",
   "ArtifactNameRequired": "Artifact Name is required.",
+  "ArtifactNotFromLinkedDefinition": "The build {0} does not belong to the linked build definition {1}",
```
Comment (GitHubSriramB) [inline src/Misc/layoutbin/en-US/strings.json]: let's name this as RMBuildNotFromLinkedDefinition

File: src/Agent.Worker/Release/Artifacts/BuildArtifact.cs [hunk 5]
```diff
@@ -241,5 +255,33 @@ private bool Match(ServerBuildArtifact buildArtifact, ArtifactDefinition artifac
                 executionContext.Warning(StringUtil.Loc("RMArtifactTypeNotSupported", buildArtifact.Resource.Type));
             }
         }
+
+        private void EnsureVersionBelongsToLinkedDefinition(ArtifactDefinition artifactDefinition, BuildHttpClient buildClient, XamlBuildHttpClient xamlBuildClient)
```
Comment (GitHubSriramB) [inline src/Agent.Worker/Release/Artifacts/BuildArtifact.cs]: is there a way we can do this for other artifact types as well?

File: src/Misc/layoutbin/en-US/strings.json (hunk 4 above)
Comment (satinder-kharea) [inline src/Misc/layoutbin/en-US/strings.json]: done.

File: src/Agent.Worker/Release/Artifacts/BuildArtifact.cs (hunk 5 above)
Comment (satinder-kharea) [inline src/Agent.Worker/Release/Artifacts/BuildArtifact.cs]: We are already doing this for custom artifact type.

**Pull request merged**